sound/soc/intel/boards/skl_hda_dsp_generic.c

```diff
@@ -150,6 +152,28 @@ static int skl_hda_fill_card_info(struct snd_soc_card *card,
 		}
 	}
 
+	if (!ctx->bt_offload_present) {
+		/* remove last link since bt audio offload is not supported */
+		num_links -= BT_DAI_COUNT;
+	} else {
+		if (codec_mask == IDISP_CODEC_MASK)
+			bt_link = &skl_hda_be_dai_links[IDISP_DAI_COUNT + DMIC_DAI_COUNT];
+		else
+			bt_link = &skl_hda_be_dai_links[IDISP_DAI_COUNT + HDAC_DAI_COUNT + DMIC_DAI_COUNT];
+
+		/* complete the link name and dai name with SSP port number */
+		bt_link->name = devm_kasprintf(dev, GFP_KERNEL, bt_link->name,
+					       ctx->ssp_bt);
+		if (!bt_link->name)
+			return -ENOMEM;
+
+		bt_link->cpus->dai_name = devm_kasprintf(dev, GFP_KERNEL,
+							 bt_link->cpus->dai_name,
```

Comment:
The dai_name comes from topology and in a form of a magic string also?

Reply:
The name is from here:

SND_SOC_DAILINK_DEF(bt_offload_pin,
	DAILINK_COMP_ARRAY(COMP_CPU("SSP%d Pin")));